sap-service-operator/src/main/java/com/sap/cloud/environment/servicebinding/FileSystemWatcherCache.java

```diff
@@ -84,12 +88,28 @@ public synchronized List<ServiceBinding> getServiceBindings( @Nonnull final Coll
     {
         removeOutdatedWatchKeys(directories);
         removeOutdatedServiceBindings(directories);
-        return directories
-            .stream()
-            .peek(this::renewCachedServiceBindingIfNeeded)
-            .map(cachedServiceBindings::get)
-            .filter(Objects::nonNull)
-            .collect(Collectors.toList());
+
+        try {
+            return directories
+                .stream()
+                .peek(this::renewCachedServiceBindingIfNeeded)
+                .map(cachedServiceBindings::get)
+                .filter(Objects::nonNull)
+                .collect(Collectors.toList());
+        }
+        catch( final Exception e ) {
```

Comment:
This looks like a very broad `catch`. Would it be possible to limit this to the concrete exception, or moving the exception handling to the stream operations above (if it's readable)?